lib/auth/userloginstate/generator.go

```diff
@@ -120,6 +140,13 @@ func (g *Generator) Generate(ctx context.Context, user types.User) (*userloginst
 		return nil, trace.Wrap(err)
 	}
 
+	if modules.GetModules().Features().Cloud {
```

Comment:
Maybe do the cloud check (which accesses a global under lock) only in the check and set defaults, and from there on just check if your `usageEvents` client is not nil?